cmd/command_argument_validator.go

```diff
@@ -0,0 +1,25 @@
+package cmd
+
+import "fmt"
+
+type CommandArgumentValidator struct {
+	required int
+	optional int
+}
+
+func (c *CommandArgumentValidator) validate(args []string) (err error) {
+	argCount := len(args)
+
+	expectedCount := fmt.Sprintf("exactly %d", c.required)
+	if (c.optional > 0) {
+		expectedCount = fmt.Sprintf("between %d and %d", c.required, c.required+c.optional)
+	}
+
+	if argCount > c.required+c.optional {
+		err = fmt.Errorf("Error: too many arguments (expected %s, got %d)\n", expectedCount, len(args))
```

Comment:
You can re-use `argCount` in both places here?

Reply:
done

Reply:
i need coffee